eventlog/tests.py

```diff
@@ -0,0 +1,71 @@
+from django.contrib.auth.models import User
+from django.test import TestCase
+from django.utils import timezone
+
+from roster.models import Site, Period, ClusiveUser
+from eventlog.models import Session, Event
+
+# TODO copied from roster tests. Is there a central place that some of this setup could be done?
+def set_up_test_sites():
+    Site.objects.create(name="CAST Collegiate", city="Wakefield", state_or_province="MA", country="USA").save()
+
+def set_up_test_periods():
+    cast_collegiate = Site.objects.get(name="CAST Collegiate")
+    Period.objects.create(name="Universal Design For Learning 101", site=cast_collegiate).save()
+
+def set_up_test_users():
+    user_1 = User.objects.create_user(username="user1", password="password1")
+    user_1.save()
+    cuser_1 = ClusiveUser.objects.create(anon_id="Student1", user=user_1, role='ST')
+    cuser_1.periods.add(Period.objects.get(name="Universal Design For Learning 101"))
+    cuser_1.save()
+
+class EventlogTestCase(TestCase):
+    def setUp(self):
+        set_up_test_sites()
+        set_up_test_periods()
+        set_up_test_users()
+
+    def test_session_created(self):
+        login = self.client.login(username='user1', password='password1')
+        self.assertTrue(login)
+        session = Session.objects.all()[0]
```

Comment:
Is there a more reliable way to do this (get the session associated with the login)? The assumption that the first object will be the associated session may be true in the testing context right now, but I worry about possible issues further down the line. I'd suggest doing something like sorting the queryset by the session creation date and getting the latest if there's no way to retrieve the associated session event based on information from the login.

I would also use `all().first()` in place of `all()[0]` when doing these kind of queries, as described at https://docs.djangoproject.com/en/2.2/ref/models/querysets/#first

Reply:
Changed to querying for most recent loginsession.